### ISSUE TITLE
Firewall rules for Build Agents

### ISSUE THREAD

**Issue by vdevappa** [MicrosoftDocs/vsts-docs#1905]
By default service fabric blocks all incoming ports. Which ports and IP addresses do I need to open for my Service Fabric cluster so the build works? Currently it only works if I open all ports and IPs. I tried opening specific ports, but the build agent fails to connect.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 47625b86-abd5-ee15-6d61-8bcf81861b96
* Version Independent ID: be9a6174-a69b-7aed-d584-5f602fcc5eb4
* Content: [Build and Release Agents - Azure Pipelines & TFS](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/agents?view=vsts)
* Content Source: [docs/pipelines/agents/agents.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/pipelines/agents/agents.md)
* Product: **devops**
* GitHub Login: @andyjlewis
* Microsoft Alias: **alewis**

**Comment by vtbassmatt** [comment]
@TingluoHuang the agents don't need any incoming ports, right? They open a long poll socket to the server.

**Comment by TingluoHuang** [comment]
@vtbassmatt yes, the agent only needs to reach out to the internet.

**Comment by vtbassmatt** [comment]
@vdevappa can you explain your scenario a bit more? If Service Fabric needs to know outbound ports, it should be 443 (HTTPS) and nothing more. No inbound ports are required.

**Comment by vdevappa** [comment]
Thanks for the replies.

I finally managed to get this working. I got the IP range from the XML file for my Central region (DevOps) and I only had to open port 19000. The mistake was trying to give that 19000 port number in both the source and destination port ranges. The correct firewall setting is shown here. 

The IP addresses being updated every week is hard. I wish there was just a static list of IPs for these build agents.

![image](https://user-images.githubusercontent.com/22800346/46216930-a7faf380-c30e-11e8-9f78-3278c3740ef3.png)
